### PR TITLE
Add documentation for JavaScript and lines_of_codes (all files)

### PR DESCRIPTION
**Pull request by mylinhdao**
# Add documentation for JavaScript and lines_of_codes (all files)
Closes: #262 
Closes: #220 
Closes: #255 

## Description
- Language documentation for JS
- Mention that lines_of_code is calculated for all files
- Move all language docs into one folder
- Add more tests for JS
## Definition of Done

A PR is only ready for merge once all the following acceptance criteria are fulfilled:
* [ ]  All TODOs related to this PR have been closed
* [x]  There are automated tests for newly written code and bug fixes
* [x]  All bugs discovered while working on this PR have been submitted as issues (if not already an open issue)
* [x]  Documentation ([README.md](https://github.com/MaibornWolff/metric-gardener/blob/main/README.md), [UPDATE_GRAMMARS.md](https://github.com/MaibornWolff/metric-gardener/blob/main/UPDATE_GRAMMARS.md)) has been updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,8 @@ The number of class definitions inside a file, also counting for enums, interfac
 **lines_of_code**<br>
 The total number of lines of a file, including empty lines, comments, etc.
 
+_Note: lines_of_code metric will be calculated for all input files, regardless of the languages they are written in._
```
Comment (clemens-mw) [inline README.md:78]: @ResistantBear mentioned, that we only calculate it for UTF-8 encoded files.
Do we also need to write it here?

File: UPDATE_GRAMMARS.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ If there are new syntax node types included in the grammars (e.g. because of a n
 If you want to support a completely new programming language, you have to perform additional steps after installing the grammar as a dependency of this project and before running the import script:
 
 -   Add the language, the file extension(s) of the source code files used by that language and an appropriate shortcut for that language to the enum and the maps inside [Language.ts](src%2Fparser%2Fhelper%2FLanguage.ts).
--   You also have to add the path to the `node-types.json` of the tree-sitter grammar for that language to [ImportNodeTypes.ts](src%2Fcommands%2Fimport-grammars%2FImportNodeTypes.ts).
+-   You also have to add the path to the [nodeTypesConfig.json](src/parser/config/nodeTypesConfig.json) of the tree-sitter grammar for that language to [ImportNodeTypes.ts](src%2Fcommands%2Fimport-grammars%2FImportNodeTypes.ts).
```
Comment (clemens-mw) [inline UPDATE_GRAMMARS.md:17]: I think it was correct before:
this refers to the `node-types.json` of the grammar in `node_modules`

maybe we can improve the phrasing

File: UPDATE_GRAMMARS.md
Comment (clemens-mw) [inline UPDATE_GRAMMARS.md:1]: should we also move this to the `docs` directory an refer from the README?
then we would need to rename the `language-specific-doc` directory

File: resources/javascript/functions-and-methods.js [hunk 3]
```diff
@@ -12,6 +13,18 @@ class FunctionsAndMethods {
...
+    }
+    set latest(age) {
+        this.#age = age;
+    }
+    *more(){
+        let index = 0;
+        while (true) {
+            yield index++;
+    }
+}
```
Comment (clemens-mw) [inline resources/javascript/functions-and-methods.js:27]: should add complexity counting to match the whole file
also add space after *more()
;)
```suggestion
    get latest() {    // complexity +1
        return 3;
    }
    set latest(age) {    // complexity +1
        this.#age = age;
    }
    *more() {    // complexity +1
        let index = 0;
        while (true) {    // complexity +1
            yield index++;
    }
}
```

**Review by clemens-mw — CHANGES_REQUESTED** [review]
looks good in total :)

File: UPDATE_GRAMMARS.md (hunk 2 above)
Comment (ResistantBear) [reply UPDATE_GRAMMARS.md:17]: Yes, this change is wrong. The `nodeTypesConfig.json` is actually already mentioned in line 23.

**Review by clemens-mw — APPROVED** [review]

**Pull request merged**